File: psra/repository/src/main/java/by/arhor/psra/repository/UserRepository.java
package by.arhor.psra.repository;

import java.util.Optional;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

import by.arhor.psra.repository.model.User;

public interface UserRepository extends MongoRepository<User, String> {

	@Query("{'username' : '?0'}")
	Optional<User> findByUsername(String username);

}

File: psra/repository/src/main/scala/by/arhor/psra/repository/UserRepository.scala
package by.arhor.psra.repository

import java.util.Optional

import by.arhor.psra.repository.model.User
import org.springframework.data.mongodb.repository.{MongoRepository, Query, ReactiveMongoRepository}

trait UserRepository extends MongoRepository[User, String] {

	@Query("{'username' : '?0'}")
	def findByUsername(username: String): Optional[User]

}
